File: framework/languages/java/samples/java/Bingo/src/main/java/systems/zlink/samples/bingo/server/play/bingoroomspots/BingoNotificationPublisher.java
package systems.zlink.samples.bingo.server.play.bingoroomspots;

import java.util.List;
import java.util.concurrent.CompletionStage;

public final class BingoNotificationPublisher {
    public CompletionStage<Void> publishWinnerAsync(
        BingoWinnerSink client,
        List<String> winners,
        String roomId) {
        return client.publishWinnerAsync(String.join(",", winners), roomId);
    }
}

File: framework/languages/java/samples/kotlin/Bingo/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/play/bingoroomspots/BingoNotificationPublisher.kt
package systems.zlink.samples.kotlin.bingo.server.play.bingoroomspots

import systems.zlink.samples.kotlin.bingo.server.play.actors.PlayerActor
import systems.zlink.samples.kotlin.bingo.shared.contracts.BingoWinner

class BingoNotificationPublisher {
    fun publishWinner(actor: PlayerActor, winners: List<String>, roomId: String) {
        actor.context()
            .boundSession()
            .send(BingoWinner(winners.joinToString(",")))
            .packetName("BingoWinner")
            .metadata("roomId", roomId)
            .submitAsync()
    }
}
